openupgradelib/openupgrade.py

```diff
@@ -834,6 +837,48 @@ def reactivate_workflow_transitions(cr, transition_conditions):
             (condition, transition_id))
 
 
+def log():
+    """Decorator for automatic logging of the executed method.
+    :param custom_logger: for defining an specific logger
+    :param details: if True, more details are given in the log"""
+    def wrap(func, custom_logger=None, details=False):
+        def wrapped_function(*args, **kwargs):
+            msg = "Executing method %s" % func.__name__
+            if details:
+                if args:
+                    msg += " with args %s" % str(args)
+                if kwargs:
+                    if args:
+                        msg += " and kwargs %s" % str(kwargs)
+                    else:
+                        msg += " with kwargs %s" % str(kwargs)
+            custom_logger.info(msg)
+            func(*args, **kwargs)
+        if custom_logger is None:
+            custom_logger = logger
+        return wrapped_function
+    return wrap
+
+
+def logged_progress(func, iterator, title="", args=None, kwargs=None):
+    """Log the progress of a method that it's run across the elements of an
+    iterator. The passed method should have the individual element to treat
+    as the first argument. It can have more arguments that are passed through.
+    """
+    widgets = [title, progressbar.Percentage(), ' ', progressbar.Bar(),
+               ' ', progressbar.ETA()]
+    if args is None:
+        args = []
+    if kwargs is None:
+        kwargs = {}
+    pbar = progressbar.ProgressBar(widgets=widgets,
+                                   maxval=len(iterator)).start()
+    for i, element in enumerate(iterator):
+        pbar.update(i)
+        func(element, *args, **kwargs)
+    pbar.finish()
```

Comment:
Use progressbar 2.3 feature:

```python
    pbar = progressbar.ProgressBar(widgets=widgets)
    for element in pbar(iterator):
        func(element, *args, **kwargs)
```

It's much nicer, especially for this straightforward case.

Also, add a space after the title.